datalad/metadata/aggregate.py

```diff
@@ -186,16 +193,130 @@ def _extract_metadata(agginto_ds, aggfrom_ds, db, to_save):
     db[aggfrom_ds.path] = agginfo
 
     if not objid:
-        dsmeta = contentmeta = None
         # this is no error, there is simply no metadata whatsoever
         return False
 
-    # if there is any chance for metadata
-    # obtain metadata for dataset and content
+    # shorten to MD5sum
+    objid = md5(objid.encode()).hexdigest()
+
+    # assemble info on the metadata extraction and storage
+    #               label  type      targetds    storage method
+    metasources = {'ds': {'type': 'dataset', 'targetds': agginto_ds, 'dumper': json_py.dump}}
```

Comment:
Now might be the time to introduce a flag to make this also use compressed+annex by default.